src/Umbraco.Cms.Integrations.Commerce.Shopify/Client/src/modal/shopify-products-modal.element.ts

```diff
@@ -0,0 +1,284 @@
+                status: lstData[i].find(x => x.columnAlias == "status")?.value,
+                tags: lstData[i].find(x => x.columnAlias == "tags")?.value,
+                variants: [],
+                image: {
+                    src: ""
+                }
+            }
+
+            productList.push(dto);
+        }
+
+        return productList;
+    }
+
+    private async _showError(message: string) {
+        const notificationContext = await this.getContext(UMB_NOTIFICATION_CONTEXT);
+        notificationContext?.peek("danger", {
+            data: { message },
+        });
+    }
+
+    #createTableItems(products: Array<ProductDtoModel>) {
+		this._tableItems = products.map((product) => {
+			return {
+                id: product.id.toString(),
+				icon: 'icon-book-alt-2',
+                data: [{
+                    columnAlias: "productName",
+                    value: product.title,
+                },
+                {
+                    columnAlias: "vendor",
+                    value: product.vendor,
+                },
+                {
+                    columnAlias: "status",
+                    value: product.status,
+                },
+                {
+                    columnAlias: "tags",
+                    value: product.tags,
+                },
+                {
+                    columnAlias: "sku",
+                    value: product.variants.map(v => v.sku).join(","),
+                },
+                {
+                    columnAlias: "barcode",
+                    value: product.variants.map(v => v.barcode).join(","),
+                },
+                {
+                    columnAlias: "price",
+                    value: product.variants[0].price,
+                },
+            ]
+            }
+		});
+	}
+
+    _onSubmit() {
+        if(this._numberOfSelection > 2){
+            this._showError("Cannot select more than 2 products."!);
+        }else{
+            if(this._modalSelectedProducts.length == 0){
+                this._rejectModal();
+            }
+
+            this.value = {productList: this._modalSelectedProducts};
+            this._submitModal();
+        }
+    }
+
+    render() {
+        return html`
+            <umb-body-layout>
+                <uui-box headline=${this.data!.headline}>
+                    ${this._loading ? html`<div class="center"><uui-loader></uui-loader></div>` : ""}
+                    <umb-table 
+                        .config=${this._tableConfig} 
+                        .columns=${this._tableColumns} 
+                        .items=${this._tableItems}
+                        .selection=${this._selection}
+                        @selected="${this.#onSelected}"
+                        @deselected="${this.#onDeselected}"></umb-table>
+                    
+                </uui-box>
+
+                <div class="maximum-selection">
+                    <span>
+                        Add up to 2 items(s)
```

Comment:
Also hardcoded to demonstrate the feature of selecting a limited number of products